src/components/TrackTile/icons/indicator/index.tsx

```diff
@@ -4,28 +4,64 @@ import { SvgProps } from 'react-native-svg';
 import { useIcons } from '../../../BrandConfigProvider';
 
 type IndicatorProps = {
+  /**
+   * @property The name of the icon to use. Can be an id for dynamic icons or a static value.
+   */
   name: string;
   color?: ColorValue;
   scale?: number;
-  CustomIcon?: React.ComponentType<SvgProps>;
+  /**
+   * @property A icon name to fallback to if `name` is not found
+   */
+  fallbackName: string;
 };
 
 export const INDICATOR_HEIGHT = 20;
 
 const Indicator = ({
   name,
   scale = 1,
-  CustomIcon,
+  fallbackName,
   ...rest
 }: IndicatorProps) => {
   const Icons = useIcons();
+  const CustomIcon = Icons[name] ?? Icons[fallbackName];
 
   const svgProps: SvgProps = {
     width: '100%',
     height: '100%',
     ...rest,
   };
 
+  const NamedIcons: Record<string, React.ReactNode> = {
+    apple: <Icons.Apple {...svgProps} />,
+    broccoli: <Icons.Broccoli {...svgProps} />,
+    'book-open': <Icons.BookOpen {...svgProps} />,
+    check: <Icons.Check {...svgProps} />,
+    chroma: <Icons.Chroma {...svgProps} />,
+    'clock-forward': <Icons.ClockForward {...svgProps} />,
+    cocktail: <Icons.Alcohol {...svgProps} />,
+    droplet: <Icons.Droplet {...svgProps} />,
+    'fast-food': <Icons.Burger {...svgProps} />,
+    flossing: <Icons.Floss {...svgProps} />,
+    heart: <Icons.Heart {...svgProps} />,
+    moon: <Icons.Moon {...svgProps} />,
+    music: <Icons.Music {...svgProps} />,
+    plant: <Icons.Plant {...svgProps} />,
+    'play-circle': <Icons.PlayCircle {...svgProps} />,
+    'plus-circle': <Icons.PlusCircle {...svgProps} />,
+    running: <Icons.Running {...svgProps} />,
+    sitting: <Icons.Sitting {...svgProps} />,
+    smile: <Icons.Smile {...svgProps} />,
+    soda: <Icons.Soda {...svgProps} />,
+    star: <Icons.Star {...svgProps} />,
+    target: <Icons.Target {...svgProps} />,
+    walking: <Icons.Steps {...svgProps} />,
+    water: <Icons.Droplet {...svgProps} />,
```

Comment:
Heads up @aymanenadi just added two icons to this list so this refactor might accidentally remove them.